tools/generate_report.py

```diff
@@ -0,0 +1,206 @@
+# Arguments and Paths
+num_args = len(sys.argv)
+
+if num_args < 1 or num_args > 4:
+    error('Unknown number of arguments')
+
+# relative paths to report directories from ROOT
+ROOT = 'data/reports'
+os.makedirs(ROOT, exist_ok = True)
+DIR_LAST_REPORT_REL = 'latest'
+OVERRIDE_LAST = True
+
+if num_args > 1: DIR_REPORT_REL = sys.argv[1]
+if num_args > 2: DIR_LAST_REPORT_REL = sys.argv[2]
+if num_args > 3: OVERRIDE_LAST = string_to_bool(sys.argv[3])
+
+DIR_REPORT = os.path.realpath(os.path.join(ROOT, DIR_REPORT_REL))
+DIR_LAST_REPORT_SYMBOLIC = os.path.join(ROOT, DIR_LAST_REPORT_REL)
+DIR_LAST_REPORT = os.path.realpath(DIR_LAST_REPORT_SYMBOLIC)
+
+REPORT_PATH = os.path.join(DIR_REPORT, 'test-status.json')
+LAST_REPORT_PATH = os.path.join(DIR_LAST_REPORT, 'test-status.json')
+
+if OVERRIDE_LAST:
+    DIR_BADGE = os.path.join('data/badges', DIR_LAST_REPORT_REL)
+    os.makedirs(DIR_BADGE, exist_ok = True)
+
+    DIR_REDIRECT = os.path.join('gh-pages', DIR_LAST_REPORT_REL)
+    os.makedirs(DIR_REDIRECT, exist_ok = True)
+
+################################################################################
+# Read current and previous test-status
+with open(REPORT_PATH, 'r') as f:
+    REPORT = json.load(f)
+
+if os.path.isfile(LAST_REPORT_PATH):
+    with open(LAST_REPORT_PATH, 'r') as f:
+        LAST_REPORT = json.load(f)
+else: # deal with the first run of this script
+    LAST_REPORT = {'pkgs': {}, 'hash': 'Unknown'}
+
+REPO = REPORT['repo']
+
+################################################################################
+# Generate report.md and test-status-diff.json
+REPORT_DIFF = {}
+REPORT_DIFF['current'] = REPORT_PATH.split('data/')[1]
+REPORT_DIFF['last'] = LAST_REPORT_PATH.split('data/')[1]
+REPORT_DIFF['total'] = REPORT['total']
+REPORT_DIFF['failure'] = REPORT['failure']
+REPORT_DIFF['success'] = REPORT['success']
+REPORT_DIFF['skipped'] = REPORT['skipped']
+
+with open(DIR_REPORT+'/report.md', 'w') as f:
+    # Header
+    f.write('# Package Evaluation Report\n\n')
+    f.write('## Job Properties\n\n')
+    f.write('*Testing:* [%s](%s) vs [%s](%s)\n\n' % (
+        REPORT_DIFF['current'].split('reports/')[1].split('/test-status.json')[0],
+        os.path.join(REPO, 'blob/data', REPORT_DIFF['current']),
+        REPORT_DIFF['last'].split('reports/')[1].split('/test-status.json')[0],
+        os.path.join(REPO, 'blob/data', REPORT_DIFF['last'])
+        ))
+    f.write('*Generated by Workflow:* %s\n\n' % REPORT['workflow'])
+    f.write('In total, %d packages were tested, out of which %d succeeded, %d failed and %d were skipped.\n\n' % (REPORT['total'], REPORT['success'], REPORT['failure'], REPORT['skipped']))
+
+    PKGS = REPORT['pkgs']
+    LAST_PKGS = LAST_REPORT['pkgs']
+
+    ############################################################################
+    # New Packages
+    PKGS_NEW = [pkg for pkg in PKGS.keys() if
+        not pkg in LAST_PKGS.keys()]
```

Comment:
I think you can just do
```suggestion
    PKGS_NEW = PKGS.keys() - LAST_PKGS.keys()
```